x/twap/types/utils.go

```diff
@@ -6,36 +6,35 @@ import (
 	"time"
 
 	sdk "github.com/cosmos/cosmos-sdk/types"
-
-	"github.com/osmosis-labs/osmosis/v12/osmoutils"
 )
 
 var MaxSpotPrice = sdk.NewDec(2).Power(128).Sub(sdk.OneDec())
 
 // GetAllUniqueDenomPairs returns all unique pairs of denoms, where for every pair
-// (X, Y), X >= Y.
+// (X, Y), X > Y.
 // The pair (X,Y) should only appear once in the list
+// Panics if finds duplicate pairs.
 //
 // NOTE: Sorts the input denoms slice.
 func GetAllUniqueDenomPairs(denoms []string) ([]string, []string) {
 	// get denoms in descending order
-	sort.Strings(denoms)
-	reverseDenoms := osmoutils.ReverseSlice(denoms)
+	sort.Slice(denoms, func(i, j int) bool {
+		return denoms[i] > denoms[j]
+	})
```

Comment:
Don't really get why we changed this / don't think its worth doing. I'd take the prior easier to read situation, this is really not relevant for performance

Reply:
https://github.com/osmosis-labs/osmosis/pull/2765/files

Seems like the preferrable approach for this fn, perhaps we merge this, and then that one?

Reply:
So what we should do if we care about performance is use the normal sort, then reverse indices for GT and LT

Reply:
I think we can instead merge #2765

Then, merge `main` into this PR. That would allow us to use `sort.Strings` here.

However, I'm not feeling strongly about this change and can revert it if you don't like. The performance benefit here is negligible since there are never many denoms